src/compiler/emitter.ts

```diff
@@ -4716,6 +4719,9 @@ if (typeof __param !== "function") __param = function (paramIndex, decorator) {
                     write(unaliasedModuleNames.join(", "));
                 }
                 write("], function (require, exports");
+                if (compilerOptions.noEmitHelpers) {

```

Comment:
But you would know if a `--module` argument has been passed wouldn't you? That's what I was trying to do with the above line:

```js
write(", \"__extends\"");
```

I was trying to add `__extends` as a module to the amd pattern. So if we are able to deduce that we are in a module then couldn't we use the `tslib.extends` pattern and use `__extends` otherwise. How's that sound?

Example taken from #1350

```js
// commonjs
var __extends = require('./tslib').extends;
var Mammal = require('./mammal');
var Human = (function (_super) {
    __extends(Human, _super);
    function Human() {
        _super.apply(this, arguments);
    }
    return Human;
})(Mammal);

// amd
define(["require", "exports", "tslib", './mammal'], function (require, exports, tslib, Mammal) {
    var Human = (function (_super) {
        tslib.extends(Human, _super);
        function Human() {
            _super.apply(this, arguments);
        }
        return Human;
    })(Mammal);
    return Human;
});

// script
var __extends = this.__extends || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    __.prototype = b.prototype;
    d.prototype = new __();
};
var Animal = (function () {
    function Animal() {
    }
    return Animal;
})();
var Mammal = (function (_super) {
    __extends(Mammal, _super);
    function Mammal() {
        _super.apply(this, arguments);
    }
    return Mammal;
})(Animal);
var Human = (function (_super) {
    __extends(Human, _super);
    function Human() {
        _super.apply(this, arguments);
    }
    return Human;
})(Mammal);
```